actionpack/lib/action_dispatch/middleware/cookies.rb

```diff
@@ -369,6 +375,10 @@ def handle_options(options) # :nodoc:
 
         options[:path] ||= "/"
 
+        if COOKIES_SAME_SITE_COOKIES_OPTIONS.include?(request.cookies_same_site_protection)
+          options[:same_site] ||= request.cookies_same_site_protection.to_s.capitalize
+        end
+
```

Comment:
If someone makes a typo in the config value, the SameSite option will be ignored. [Rack validates](https://github.com/rack/rack/pull/1051) allowed `:same_site` values. Can we rely on that?

Reply:
I believe there's also the value of `None`

See:

https://tools.ietf.org/html/draft-ietf-httpbis-rfc6265bis-03

```
samesite-value    = "Strict" / "Lax" / "None"
```

https://www.chromestatus.com/feature/5088147346030592
https://www.chromestatus.com/feature/5633521622188032 - `None` cookies have to be secure

Reply:
Note [this is necessary for cross-domain iframe and AJAX requests](https://makandracards.com/makandra/71018-samesite-cookies#what-this-means-for-web-development) that rely on cookies to be sent.

Reply:
Don't know if you'd wanna note that [it also is incompatible with a whole host of clients](https://www.chromium.org/updates/same-site/incompatible-clients).